src/vivarium/framework/results/context.py

```diff
@@ -103,7 +103,7 @@ def add_observation(
 
     def gather_results(
         self, population: pd.DataFrame, event_name: str
-    ) -> Generator[Optional[pd.DataFrame], None, None]:
+    ) -> Generator[Tuple[Optional[pd.DataFrame], Optional[str]], None, None]:
```

Comment:
Rajan: Try yielding only None w/ typing `-> Generator[Optional[Tuple[pd.Dataframe, str]], None, None]`

Reply:
Not a clean way to do this